tensorboard/webapp/metrics/views/card_renderer/scalar_card_container.ts

```diff
@@ -611,10 +612,28 @@ export class ScalarCardContainer implements CardRenderer, OnInit, OnDestroy {
   onTimeSelectionChanged(
     newTimeSelectionWithAffordance: TimeSelectionWithAffordance
   ) {
+    let start = newTimeSelectionWithAffordance.timeSelection.start;
+    const {minStep, maxStep} = this.minMaxSteps$.getValue();
+
+    start = {
+      step:
+        start.step < minStep
+          ? minStep
+          : start.step > maxStep
+          ? maxStep
+          : start.step,
+    };
+
+    // TODO: bound end step when range selection is implemented.
+
+    const newTimeSelection = {
+      start,
+      end: newTimeSelectionWithAffordance.timeSelection.end,
+    };
+
```

Comment:
This does break range selection when linked time is enabled.

![image](https://user-images.githubusercontent.com/78179109/192060656-5ee935c3-37e0-4169-8f03-808c04591ef4.png)

I was hopeful it wouldn't because of the existing bounding logic used to compute the `minMaxStep` however that logic is only used to compute the fob position. I'm not sure why this logic needs to be different though

```suggestion
    const {minStep, maxStep} = this.minMaxSteps$.getValue();
    const {startStep, endStep} = maybeClipLinkedTimeSelection(newTimeSelectionWithAffordance.timeSelection, minStep, maxStep);
    const newTimeSelection = {
      start: {step: startStep},
      end: endStep ? {step: endStep} : null
    };
```

My PR changing the clipping logic makes this experience a little bit better but I think it is fine even in its current form.

Before #5938
Step  Selection
![84ee8f0d-e08b-4f25-ae1b-f925b9a0abec](https://user-images.githubusercontent.com/78179109/192061757-be387b43-5cdb-495b-9044-5ddbe0656ec5.gif)

Linked Time + Time Selection
![02d4a32c-3f3e-4201-b075-5807845d79a7](https://user-images.githubusercontent.com/78179109/192061811-25446bee-1a8c-4ccf-9e3f-5235c07ba1c1.gif)

After #5938
Step Selection
![2750b40b-3745-46e7-9001-e735b4ca5fd9](https://user-images.githubusercontent.com/78179109/192061544-0686c5b5-0da3-456e-8fe8-658381145896.gif)

Linked Time + Time Selection
![60db7e06-48af-4e61-94d8-8c7544ed7c0e](https://user-images.githubusercontent.com/78179109/192061637-5088bb85-70f2-48d5-82e8-e32d3429b085.gif)

Reply:
I like it. Done.